openhtf/exe/__init__.py

```diff
@@ -209,3 +215,7 @@ def _ExecuteTestPhases(self, executor):
         break
     else:
       self._test_state.SetStateFinished()
+    # Run teardown function.
+    if self._teardown_function:
+      self._test_state.pending_phases.insert(0, self._teardown_function)
```

Comment:
removed pending phases, added check for empty list around the pop in phase executor